covfee/client/journey/node_loader.tsx

```diff
@@ -297,6 +299,17 @@ export const NodeLoader: React.FC<Props> = (props: Props) => {
       })
   }
 
+  const handleTaskProgressSubmit = (progress: number) => {
+    submitProgress(progress)
+      .then((data: any) => {
+        setNodeProgress(progress)
+      })
+      .catch((error) => {
+        myerror("Error updating task progress.", error)
+        setNodeProgress(progress)
```

Comment:
If there was an error, would it be better to not try to set the progress again but just let it pass?

Reply:
note that `setNoteProgress` is about setting the react state so that progress is displayed as desired in the user interface(s); just in case, it's not a "try again". What's problematic is that there would be a lack of sync between the server state, and the browser state, but 1) I replicated the pattern related to the "submit response" logic (see https://github.com/TUDelft-SPC-Lab/covfee/blob/d5a80c66b0ed0222551aa48f77eca760829fadb5/covfee/client/journey/node_loader.tsx#L296 ), and; 2) worse case, the app will recover quickly in the next call to handleTaskProgressSubmit which will happen even in a refresh, which the user would be compelled to do if things look weird (I assume if it can't communicate to the server, then other major issues will start to occur)